package/MDAnalysis/lib/c_distances.pyx

```diff
@@ -315,3 +317,290 @@ def contact_matrix_pbc(coord, sparse_contacts, box, cutoff):
+            indices[p] = i
+            p += 1
+           
+            if lo_y:
+                # if X&Y, edge piece
+                for j in range(3):
+                    output[p, j] = coord[j] + shiftX[j] + shiftY[j]
+                indices[p] = i
+                p += 1
+                
+                
+                if lo_z:
+                    # if X&Y&Z, corner piece
+                    for j in range(3):
+                        output[p, j] = coord[j] + shiftX[j] + shiftY[j] + shiftZ[j]
+                    indices[p] = i
+                    p += 1
+                    
+                    
+                elif hi_z:
+                    for j in range(3):
+                        output[p, j] = coord[j] + shiftX[j] + shiftY[j] - shiftZ[j]
+                    indices[p] = i
+                    p += 1
+                    
+
+            elif hi_y:
+                for j in range(3):
+                    output[p, j] = coord[j] + shiftX[j] - shiftY[j]
+                indices[p] = i
+                p += 1
+                
+                
+                if lo_z:
+                    for j in range(3):
+                        output[p, j] = coord[j] + shiftX[j] - shiftY[j] + shiftZ[j]
+                    indices[p] = i
+                    p += 1
+                    
+                    
+                elif hi_z:
+                    for j in range(3):
+                        output[p, j] = coord[j] + shiftX[j] - shiftY[j] - shiftZ[j]
+                    indices[p] = i
+                    p += 1
+                    
+                
+            if lo_z:
+                for j in range(3):
+                    output[p, j] = coord[j] + shiftX[j] + shiftZ[j]
+                indices[p] = i
+                p += 1
+                
+               
+            elif hi_z:
+                for j in range(3):
+                    output[p, j] = coord[j] + shiftX[j] - shiftZ[j]
+                indices[p] = i
+                p += 1
+                
+                
+        elif hi_x:
+            for j in range(3):
+                output[p, j] = coord[j] - shiftX[j]
+            indices[p] = i
+            p += 1
+            
+            
+            if lo_y:
+                for j in range(3):
+                    output[p, j] = coord[j] - shiftX[j] + shiftY[j]
+                indices[p] = i
+                p += 1
+                
+                
+                if lo_z:
+                    for j in range(3):
+                        output[p, j] = coord[j] - shiftX[j] + shiftY[j] + shiftZ[j]
+                    indices[p] = i
+                    p += 1
+                    
+                    
+                elif hi_z:
+                    for j in range(3):
+                        output[p, j] = coord[j] - shiftX[j] + shiftY[j] - shiftZ[j]
+                    indices[p] = i
+                    p += 1
+                    
+                
```

Comment:
if you can fix up the formatting it would be easier to read, there's 3 blank lines here